core/src/main/java/org/infinispan/util/RpcUtil.java

```diff
@@ -0,0 +1,30 @@
+package org.infinispan.util;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Some utility methods for remoting.
+ *
+ * @author Pedro Ruivo
+ * @since 8.1
+ */
+public class RpcUtil {
+
+   public static final long BIG_DELAY_NANO = TimeUnit.DAYS.toNanos(1);
+
+   private RpcUtil() {
+   }
+
+   public static <T> T waitFor(CompletableFuture<T> future)
```

Comment:
One more (very late) suggestion: this should be useful for anything that needs to wait on a `CompletableFuture`, so I'd move it to `CompletableFutures`.